---
title: "Phyluce"
output: html_notebook
---

This notebook will document my work with the package [Phyluce](http://phyluce.readthedocs.io/en/latest/index.html) by Brant Faircloth.


# Installation and Pre-Processing
1. Installed [Anaconda with Python 2.7](https://docs.anaconda.com/anaconda/install/)

2. Installed Phyluce (used extra advice from [here](https://github.com/faircloth-lab/phyluce/issues/57) to create a new environment)
```{bash}
M053-303-26932:~ cran5048$ conda create --name phyluce
Solving environment: done


## Package Plan ##

  environment location: /anaconda2/envs/phyluce


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use:
# > source activate phyluce
#
# To deactivate an active environment, use:
# > source deactivate
#
M053-303-26932:~ cran5048$ source activate phyluce
(phyluce) M053-303-26932:~ cran5048$ conda install phyluce
Solving environment: done


M053-303-26932:~ cran5048$ conda install phyluce
Downloading and Extracting Packages
gblocks 0.91b: ########################################################################## | 100% 
raxml 8.0.19: ########################################################################### | 100% 
abyss 1.3.7: ############################################################################ | 100% 
pandas 0.13.0: ########################################################################## | 100% 
dateutil 2.4.1: ######################################################################### | 100% 
pysam 0.7.7: ############################################################################ | 100% 
bx-python 0.7.1: ######################################################################## | 100% 
biopython 1.63: ######################################################################### | 100% 
numpy 1.7.1: ############################################################################ | 100% 
mafft 7.130: ############################################################################ | 100% 
trimmomatic 0.32: ####################################################################### | 100% 
pip 9.0.3: ############################################################################## | 100% 
wheel 0.31.0: ########################################################################### | 100% 
bedtools 2.18.1: ######################################################################## | 100% 
bwa 0.7.7: ############################################################################## | 100% 
python 2.7.14: ########################################################################## | 100% 
illumiprocessor 2.0.7: ################################################################## | 100% 
samtools 0.1.19: ######################################################################## | 100% 
muscle 3.8.31: ########################################################################## | 100% 
setuptools 39.0.1: ###################################################################### | 100% 
pyvcf 0.6.4: ############################################################################ | 100% 
condaflmake 3.8.2: ###################################################################### | 100% 
dendropy 3.12.0: ######################################################################## | 100% 
bowtie 1.1.1: ########################################################################### | 100% 
trinity 2.0.6: ########################################################################## | 100% 
distribute 0.6.45: ###################################################################### | 100% 
velvet 1.2.10: ########################################################################## | 100% 
lastz 1.02.00: ########################################################################## | 100% 
phyluce 1.5.0: ########################################################################## | 100% 
picard 1.106: ########################################################################### | 100% 
gatk-lite 2.3.0: ######################################################################## | 100% 
pytz 2018.3: ############################################################################ | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done


#illumiprocessor isn't working so I removed it and reinstalled it (used force to only remove this package)
conda remove illumiprocessor --force
conda install illumiprocessor #upgrades from 2.0.7 to 2.0.8

# and install Java 8 (I have 10 installed)
brew cask install caskroom/versions/java8
#now install jenv to manage java versions, add init line to .bash_profile as suggested here https://gist.github.com/tomysmile/a9a7aee85ff73454bd57e198ad90e614
brew install jenv
#find paths to java installations
/usr/libexec/java_home -V
# add java8 to jenv
jenv add /Library/Java/JavaVirtualMachines/jdk1.8.0_162.jdk/Contents/Home
jenv versions
#set java version to 8
jenv local 1.8.0.162

#So I fixed the lines 28 and 33 in #~/anaconda2/envs/phyluce/lib/python2.7/site-packages/illumiprocessor/cli/main.py

#to add "anaconda2" in the paths there (which originally just had "anaconda")

```

3. Rename directories to reflect the sample

```{bash, evaluate=F}
for dir in */
  do
  cd $dir
  for file in *.gz; do
    cd ..
    mv $dir ${file%%_L001_R1_001.fastq.gz}
    break 1
    done
  done
  "$dir"*.gz
```

4. Copy all files into phyluce directory

```{bash, evaluate=F}
for dir in */; do
cd $dir
  for f in *.gz; do
  cp $f ../phyluce/$f
  done
cd ..
done

```

5. Cat together two individuals that were sequenced twice (once when I thought I wasn't going to include most S. ovalis in the run)

```{bash, evaluate=F}
cat sebova-sbc-R010965_S23_L001_R1_001.fastq.gz sebova-sbc-R010965_S44_L001_R1_001.fastq.gz > sebova-sbc-R010965_S2344_L001_R1_001.fastq.gz


cat sebova-sbc-R010965_S23_L001_R2_001.fastq.gz sebova-sbc-R010965_S44_L001_R2_001.fastq.gz > sebova-sbc-R010965_S2344_L001_R2_001.fastq.gz


cat sebova-sbc-R010966_S24_L001_R1_001.fastq.gz sebova-sbc-R010966_S45_L001_R1_001.fastq.gz > sebova-sbc-R010966_S2445_L001_R1_001.fastq.gz


cat sebova-sbc-R010966_S24_L001_R2_001.fastq.gz sebova-sbc-R010966_S45_L001_R2_001.fastq.gz > sebova-sbc-R010966_S2445_L001_R2_001.fastq.gz


rm sebova-sbc-R010965_S23_L001_R1_001.fastq.gz sebova-sbc-R010965_S44_L001_R1_001.fastq.gz
rm sebova-sbc-R010965_S23_L001_R2_001.fastq.gz sebova-sbc-R010965_S44_L001_R2_001.fastq.gz
rm sebova-sbc-R010966_S24_L001_R1_001.fastq.gz sebova-sbc-R010966_S45_L001_R1_001.fastq.gz
rm sebova-sbc-R010966_S24_L001_R2_001.fastq.gz sebova-sbc-R010966_S45_L001_R2_001.fastq.gz
```

# Illumiprocessor

4. Create configuration file (illumiprocessor_UCE2_config.txt)
5. Run [illumiprocessor](http://illumiprocessor.readthedocs.io/en/latest/usage.html) to trim adapters and low quality bases
```{bash, evaluate=F}
illumiprocessor --input ./ --output uce-clean --config illumiprocessor_UCE2_config.txt --cores 4

2018-04-04 18:19:59,968 - illumiprocessor - INFO - ==================== Starting illumiprocessor ===================
2018-04-04 18:19:59,990 - illumiprocessor - INFO - Version: 2.0.6
2018-04-04 18:19:59,990 - illumiprocessor - INFO - Argument --config: illumiprocessor_UCE2_config.txt
2018-04-04 18:19:59,990 - illumiprocessor - INFO - Argument --cores: 4
2018-04-04 18:19:59,990 - illumiprocessor - INFO - Argument --input: /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce
2018-04-04 18:19:59,991 - illumiprocessor - INFO - Argument --log_path: None
2018-04-04 18:19:59,991 - illumiprocessor - INFO - Argument --min_len: 40
2018-04-04 18:19:59,991 - illumiprocessor - INFO - Argument --no_merge: False
2018-04-04 18:19:59,991 - illumiprocessor - INFO - Argument --output: /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/uce-clean
2018-04-04 18:19:59,991 - illumiprocessor - INFO - Argument --phred: phred33
2018-04-04 18:19:59,991 - illumiprocessor - INFO - Argument --r1_pattern: None
2018-04-04 18:19:59,991 - illumiprocessor - INFO - Argument --r2_pattern: None
2018-04-04 18:19:59,991 - illumiprocessor - INFO - Argument --se: False
2018-04-04 18:19:59,991 - illumiprocessor - INFO - Argument --trimmomatic: /Users/cran5048/anaconda2/envs/phyluce/jar/trimmomatic.jar
2018-04-04 18:19:59,991 - illumiprocessor - INFO - Argument --verbosity: INFO
2018-04-04 18:20:00,410 - illumiprocessor - INFO - Trimming samples with Trimmomatic
Running......................................................................
2018-04-04 18:26:09,219 - illumiprocessor - INFO - =================== Completed illumiprocessor ===================

```

# Assembly

6. Create configuration file (assembly_conf.txt)
  - make this using the sample sheet in Excel to creat the paths. Need to use absolute paths to each directory
7. Run velvet
```{bash, evaluate=F}
mkdir assembly
# make a directory for log files
mkdir log
# run the assembly
phyluce_assembly_assemblo_velvet \
    --config assembly_conf.txt \
    --output ~/Basespace/UCE_Fish_2/phyluce/assembly \
    --kmer 35 \
    --subfolder split-adapter-quality-trimmed \
    --cores 4 \
    --clean \
    --log-path log

```
This took 3-4 hours to run
# Finding UCEs
8. Use LastZ to find matches to UCE probes among contigs - place them in an sql database
```{bash, evaluate=F}
mkdir FindUCE_log
# match contigs to probes
phyluce_assembly_match_contigs_to_probes \
    --contigs ./assembly/contigs \
    --probes fish-uce-500-probes.fasta \
    --output ./findUCE \
    --log-path FindUCE_log
```
This ran pretty quickly - 30 mins?

9. Check out the database
```{bash, evaluate=F}
(phyluce) M053-303-26932:FindUCE cran5048$ sqlite3 probe.matches.sqlite
SQLite version 3.22.0 2018-01-22 18:45:57
Enter ".help" for usage hints.
```

```{r}
library(DBI)
db<- dbConnect(RSQLite::SQLite(),dbname="~/Basespace/UCE_Fish_2/phyluce/FindUCE/probe.matches.sqlite")

```

```{sql connection=db, output.var="allUCEs"}

SELECT * FROM matches;




```

```{r}
rownames(allUCEs)<-allUCEs$uce

sebgoo<-allUCEs[,grep("sebgoo",names(allUCEs))]

sebova<-allUCEs[,grep("sebova",names(allUCEs))]

emblat<-allUCEs[,grep("emblat",names(allUCEs))]

embjac<-allUCEs[,grep("embjac",names(allUCEs))]

which(complete.cases(sebgoo))
which(complete.cases(sebova))
which(complete.cases(embjac))
which(complete.cases(emblat))

sebgoo<-apply(sebgoo[,1:20], FUN=as.numeric,MARGIN=2)
sebgoo_sum<-rowSums(sebgoo,na.rm=T)
names(sebgoo_sum)<-rownames(allUCEs)
sort(sebgoo_sum)

sebova<-apply(sebova[,1:17], FUN=as.numeric,MARGIN=2)
sebova_sum<-rowSums(sebova,na.rm=T)
names(sebova_sum)<-rownames(allUCEs)
sort(sebova_sum)

embjac<-apply(embjac[,1:17], FUN=as.numeric,MARGIN=2)
embjac_sum<-rowSums(embjac,na.rm=T)
names(embjac_sum)<-rownames(allUCEs)
sort(embjac_sum)

emblat<-apply(emblat[,1:15], FUN=as.numeric,MARGIN=2)
emblat_sum<-rowSums(emblat,na.rm=T)
names(emblat_sum)<-rownames(allUCEs)
sort(emblat_sum)

```
So there were `length(which(complete.cases(allUCEs)))` UCEs that got enriched in all samples :-( but not surprising as there were 72 taxa.

So I might want to back up and have it do databases for each species. Or actually I can subselect taxon sets below. But I will plow ahead and use all data for now

10. Make a list of all taxa and UCEs that I can get for the whole dataset (this is just a test prior to running the next steps)
```{bash, evaluate=F}
phyluce_assembly_get_match_counts \
    --locus-db /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/FindUCE/probe.matches.sqlite \
    --taxon-list-config /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/all_taxa_sets_conf.txt\
    --taxon-group 'all_taxa_dataset' \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa/alltaxa_dataset1_conf.txt \
    --incomplete-matrix
    
```

11. Extract all the matching data into a FASTA
```{bash, evaluate=F}
phyluce_assembly_get_fastas_from_match_counts \
    --contigs /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/assembly/contigs/ \
    --locus-db /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/FindUCE/probe.matches.sqlite \
    --match-count-output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa/alltaxa_dataset1_conf.txt \
    --incomplete-matrix /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa/alltaxa_dataset1.incomplete \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa/alltaxa_dataset1.fasta
    
```

12. Align the FASTA Reads
```{bash, evaluate=F}

phyluce_align_seqcap_align \
    --fasta /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa/alltaxa_dataset1.fasta \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa/alignments \
    --taxa 70 \
    --aligner mafft \
    --cores 4 \
    --proportion 0.05 \
    --threshold 0.05 \
    --incomplete-matrix \
    --max-divergence 0.05 \
    --min-length 50 \
```

After playing with the data I set a minimum length of 50, a proportion of 0.1 and max divergence of 0.05 and found 3 loci with intraspecific variation
51 (sebova), 1169 (sebgoo, sebova), 1296-(sebgoo)

# Backing up to Assembly

7. Run velvet with larger kmer value
```{bash, evaluate=F}
mkdir assembly
# make a directory for log files
mkdir log
# run the assembly
phyluce_assembly_assemblo_velvet \
    --config assembly_conf.txt \
    --output ~/Basespace/UCE_Fish_2/phyluce/assembly \
    --kmer 65 \
    --subfolder split-adapter-quality-trimmed \
    --cores 4 \
    --clean \
    --log-path log

```

8. Use LastZ to find matches to UCE probes among contigs - place them in an sql database
```{bash, evaluate=F}
mkdir FindUCE_log
# match contigs to probes
phyluce_assembly_match_contigs_to_probes \
    --contigs ./assembly/contigs \
    --probes fish-uce-500-probes.fasta \
    --output ./findUCE_k65 \
    --log-path FindUCE_log_k65
```


10. Make a list of all taxa and UCEs that I can get for the whole dataset (this is just a test prior to running the next steps)
```{bash, evaluate=F}
phyluce_assembly_get_match_counts \
    --locus-db /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/FindUCE_k65/probe.matches.sqlite \
    --taxon-list-config /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/all_taxa_sets_conf.txt\
    --taxon-group 'all_taxa_dataset' \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa_k65/alltaxa_dataset1_conf.txt \
    --incomplete-matrix
    
```

11. Extract all the matching data into a FASTA
```{bash, evaluate=F}
phyluce_assembly_get_fastas_from_match_counts \
    --contigs /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/assembly_k65/contigs/ \
    --locus-db /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/FindUCE_k65/probe.matches.sqlite \
    --match-count-output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa_k65/alltaxa_dataset1_conf.txt \
    --incomplete-matrix /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa_k65/alltaxa_dataset1.incomplete \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa_k65/alltaxa_dataset1.fasta
    
```


12. Align the FASTA Reads
```{bash, evaluate=F}

phyluce_align_seqcap_align \
    --fasta /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa_k65/alltaxa_dataset1.fasta \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa_k65/alignments \
    --taxa 70 \
    --aligner mafft \
    --cores 4 \
    --proportion 0.75 \
    --incomplete-matrix \
    --max-divergence 0.05 \
    --min-length 50 \
```

UCE-1020 UCE34 UCE103 UCE109 (Seb) UCE129 (Seb) UCE167 (Seb) UCE245 (Seb) UCE267 (Seb) UCE 307 (Seb) UCE 344( Sebova) UCE 418 (sebova) UCE 431 (Sebgoo) UCE 461 (Sebgoo) UCE377 (Sebova)


# Breaking it down by species
10. Make a list of all taxa and UCEs that I can get for the whole dataset (this is just a test prior to running the next steps)
```{bash, evaluate=F}
phyluce_assembly_get_match_counts \
    --locus-db /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/FindUCE_k65/probe.matches.sqlite \
    --taxon-list-config /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/all_taxa_sets_conf.txt\
    --taxon-group 'sebgoo' \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebgoo/sebgoo_dataset1_conf.txt \
    --incomplete-matrix
    
    
    phyluce_assembly_get_match_counts \
    --locus-db /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/FindUCE_k65/probe.matches.sqlite \
    --taxon-list-config /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/all_taxa_sets_conf.txt\
    --taxon-group 'sebova' \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebova/sebova_dataset1_conf.txt \
    --incomplete-matrix
    
    phyluce_assembly_get_match_counts \
    --locus-db /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/FindUCE_k65/probe.matches.sqlite \
    --taxon-list-config /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/all_taxa_sets_conf.txt\
    --taxon-group 'emblat' \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/emblat/emblat_dataset1_conf.txt \
    --incomplete-matrix
    
    phyluce_assembly_get_match_counts \
    --locus-db /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/FindUCE_k65/probe.matches.sqlite \
    --taxon-list-config /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/all_taxa_sets_conf.txt\
    --taxon-group 'embjac' \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/embjac/embjac_dataset1_conf.txt \
    --incomplete-matrix
    
```

## Sebastes Ovalis
11. Extract all the matching data into a FASTA
```{bash, evaluate=F}
phyluce_assembly_get_fastas_from_match_counts \
    --contigs /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/assembly_k65/contigs/ \
    --locus-db /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/FindUCE_k65/probe.matches.sqlite \
    --match-count-output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebova/sebova_dataset1_conf.txt \
    --incomplete-matrix /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebova/sebova_dataset1.incomplete \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebova/sebova_dataset1.fasta
    
```

Align the FASTA Reads
```{bash, evaluate=F}

phyluce_align_seqcap_align \
    --fasta /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebova/sebova_dataset1.fasta \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebova/alignments \
    --taxa 17 \
    --aligner mafft \
    --cores 4 \
    --proportion 0.75 \
    --incomplete-matrix \
    --max-divergence 0.03 \
    --min-length 50
```



Get summary stats
```{bash, evaluate=F}
phyluce_align_get_align_summary_data \
    --alignments ./sebova/alignments \
    --cores 4 \
    --log-path log
    
    2018-04-07 14:29:57,493 - phyluce_align_get_align_summary_data - INFO - ========= Starting phyluce_align_get_align_summary_data =========
2018-04-07 14:29:57,493 - phyluce_align_get_align_summary_data - INFO - Version: 1.5.0
2018-04-07 14:29:57,493 - phyluce_align_get_align_summary_data - INFO - Argument --alignments: /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebova/alignments
2018-04-07 14:29:57,493 - phyluce_align_get_align_summary_data - INFO - Argument --cores: 4
2018-04-07 14:29:57,494 - phyluce_align_get_align_summary_data - INFO - Argument --input_format: nexus
2018-04-07 14:29:57,494 - phyluce_align_get_align_summary_data - INFO - Argument --log_path: /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/log
2018-04-07 14:29:57,494 - phyluce_align_get_align_summary_data - INFO - Argument --show_taxon_counts: False
2018-04-07 14:29:57,494 - phyluce_align_get_align_summary_data - INFO - Argument --verbosity: INFO
2018-04-07 14:29:57,494 - phyluce_align_get_align_summary_data - INFO - Getting alignment files
2018-04-07 14:29:57,565 - phyluce_align_get_align_summary_data - INFO - Computing summary statistics using 4 cores
2018-04-07 14:29:58,807 - phyluce_align_get_align_summary_data - INFO - ----------------------- Alignment summary -----------------------
2018-04-07 14:29:58,807 - phyluce_align_get_align_summary_data - INFO - [Alignments] loci:	477
2018-04-07 14:29:58,807 - phyluce_align_get_align_summary_data - INFO - [Alignments] length:	79,651
2018-04-07 14:29:58,807 - phyluce_align_get_align_summary_data - INFO - [Alignments] mean:	166.98
2018-04-07 14:29:58,807 - phyluce_align_get_align_summary_data - INFO - [Alignments] 95% CI:	5.94
2018-04-07 14:29:58,807 - phyluce_align_get_align_summary_data - INFO - [Alignments] min:	55
2018-04-07 14:29:58,807 - phyluce_align_get_align_summary_data - INFO - [Alignments] max:	484
2018-04-07 14:29:58,808 - phyluce_align_get_align_summary_data - INFO - ------------------------- Taxon summary -------------------------
2018-04-07 14:29:58,808 - phyluce_align_get_align_summary_data - INFO - [Taxa] mean:		7.87
2018-04-07 14:29:58,808 - phyluce_align_get_align_summary_data - INFO - [Taxa] 95% CI:	0.21
2018-04-07 14:29:58,808 - phyluce_align_get_align_summary_data - INFO - [Taxa] min:		3
2018-04-07 14:29:58,808 - phyluce_align_get_align_summary_data - INFO - [Taxa] max:		15
2018-04-07 14:29:58,809 - phyluce_align_get_align_summary_data - INFO - ----------------- Missing data from trim summary ----------------
2018-04-07 14:29:58,809 - phyluce_align_get_align_summary_data - INFO - [Missing] mean:	10.06
2018-04-07 14:29:58,809 - phyluce_align_get_align_summary_data - INFO - [Missing] 95% CI:	0.36
2018-04-07 14:29:58,809 - phyluce_align_get_align_summary_data - INFO - [Missing] min:	0.84
2018-04-07 14:29:58,809 - phyluce_align_get_align_summary_data - INFO - [Missing] max:	22.46
2018-04-07 14:29:58,814 - phyluce_align_get_align_summary_data - INFO - -------------------- Character count summary --------------------
2018-04-07 14:29:58,814 - phyluce_align_get_align_summary_data - INFO - [All characters]	613,544
2018-04-07 14:29:58,814 - phyluce_align_get_align_summary_data - INFO - [Nucleotides]		546,272
2018-04-07 14:29:58,814 - phyluce_align_get_align_summary_data - INFO - ---------------- Data matrix completeness summary ---------------
2018-04-07 14:29:58,815 - phyluce_align_get_align_summary_data - INFO - [Matrix 50%]		267 alignments
2018-04-07 14:29:58,815 - phyluce_align_get_align_summary_data - INFO - [Matrix 55%]		180 alignments
2018-04-07 14:29:58,815 - phyluce_align_get_align_summary_data - INFO - [Matrix 60%]		180 alignments
2018-04-07 14:29:58,815 - phyluce_align_get_align_summary_data - INFO - [Matrix 65%]		119 alignments
2018-04-07 14:29:58,815 - phyluce_align_get_align_summary_data - INFO - [Matrix 70%]		68 alignments
2018-04-07 14:29:58,815 - phyluce_align_get_align_summary_data - INFO - [Matrix 75%]		29 alignments
2018-04-07 14:29:58,815 - phyluce_align_get_align_summary_data - INFO - [Matrix 80%]		29 alignments
2018-04-07 14:29:58,815 - phyluce_align_get_align_summary_data - INFO - [Matrix 85%]		9 alignments
2018-04-07 14:29:58,815 - phyluce_align_get_align_summary_data - INFO - [Matrix 90%]		5 alignments
2018-04-07 14:29:58,816 - phyluce_align_get_align_summary_data - INFO - [Matrix 95%]		1 alignments
2018-04-07 14:29:58,816 - phyluce_align_get_align_summary_data - INFO - ------------------------ Character counts -----------------------
2018-04-07 14:29:58,816 - phyluce_align_get_align_summary_data - INFO - [Characters] '-' is present 2,617 times
2018-04-07 14:29:58,816 - phyluce_align_get_align_summary_data - INFO - [Characters] '?' is present 64,655 times
2018-04-07 14:29:58,816 - phyluce_align_get_align_summary_data - INFO - [Characters] 'A' is present 148,350 times
2018-04-07 14:29:58,816 - phyluce_align_get_align_summary_data - INFO - [Characters] 'C' is present 124,944 times
2018-04-07 14:29:58,816 - phyluce_align_get_align_summary_data - INFO - [Characters] 'G' is present 124,453 times
2018-04-07 14:29:58,816 - phyluce_align_get_align_summary_data - INFO - [Characters] 'T' is present 148,525 times
2018-04-07 14:29:58,817 - phyluce_align_get_align_summary_data - INFO - ========= Completed phyluce_align_get_align_summary_data ========


```

```{bash, evaluate=F}
phyluce_align_explode_alignments \
    --alignments ./sebova/alignments \
    --output ./sebova/exploded-fastas \
    --input-format nexus \
    --by-taxon
    
```

Create a phasing config file as described [here](http://phyluce.readthedocs.io/en/latest/tutorial-two.html)

`phyluce_snp_bwa_multiple_align` was not included in the distribution. I copied it from [here](https://github.com/faircloth-lab/phyluce/blob/master/bin/snps/phyluce_snp_bwa_multiple_align)
```{bash, evaluate=F}
phyluce_snp_bwa_multiple_align \
    --config ./sebova/sebova_phasing_conf.txt \
    --output ./sebova/mapping \
    --subfolder split-adapter-quality-trimmed \
    --log-path ./sebova/bwa_log \
    --cores 4
    
```

To get this to work, I had to clone the latest git version of phyluce into my github directory and add
it to my python path.
```{bash, evaluate=F}
export PYTHONPATH=$HOME/github/phyluce

python ~/github/phyluce/bin/snps/phyluce_snp_phase_uces \
    --config ./sebova/sebova_phasing_conf.txt \
    --bams ./sebova/mapping \
    --output ./sebova/phased_reads
```

Going off on my own now, trying to call SNPs in the mapped bam files. Following guidance from [here](http://samtools.sourceforge.net/mpileup.shtml).

```{bash, evaluate=F}
(phyluce) M053-303-26932:mapping cran5048$ pwd
/Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebova/mapping
(phyluce) M053-303-26932:mapping cran5048$ find . -name "*.bam" -exec cp {} ./bams \;

#Following guidance from 
#index the fasta file at the top of the sebova directory
(phyluce) M053-303-26932:sebova cran5048$ pwd
/Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebova
(phyluce) M053-303-26932:sebova cran5048$ samtools faidx
Usage: faidx <in.fasta> [<reg> [...]]
(phyluce) M053-303-26932:sebova cran5048$ samtools faidx 
.DS_Store                   exploded-fastas/            sebova_dataset1.fasta       sebova_phasing_conf.txt
alignments/                 mapping/                    sebova_dataset1.incomplete  
bwa_log/                    phased_reads/               sebova_dataset1_conf.txt    
(phyluce) M053-303-26932:sebova cran5048$ samtools faidx sebova_dataset1.fasta

#use mpileup to compute likelihood of each genotype
(phyluce) M053-303-26932:sebova cran5048$ cd mapping
(phyluce) M053-303-26932:mapping cran5048$ cd bams
(phyluce) M053-303-26932:bams cran5048$ samtools mpileup -uf ../../sebova_dataset1.fasta *.bam | bcftools view -bvcg - > var.raw.bcf
[mpileup] 17 samples in 17 input files
<mpileup> Set max per-file depth to 470
[afs] 0:6361.738 1:634.648 2:644.429 3:656.019 4:667.827 5:679.541 6:691.061 7:702.351 8:713.395 9:724.189 10:734.735 11:745.037 12:755.103 13:764.940 14:774.556 15:783.963 16:793.170 17:802.188 18:811.029 19:819.704 20:828.226 21:836.607 22:844.859 23:852.995 24:861.028 25:868.971 26:876.836 27:884.638 28:892.388 29:900.100 30:907.787 31:915.461 32:923.134 33:930.819 34:938.527

#call the SNPs
(phyluce) M053-303-26932:bams cran5048$ bcftools view var.raw.bcf | vcfutils.pl varFilter -D100 > var.flt.vcf