---
title: "Phyluce"
output: html_notebook
---

This notebook will document my work with the package [Phyluce](http://phyluce.readthedocs.io/en/latest/index.html) by Brant Faircloth.


# Installation and Pre-Processing
1. Installed [Anaconda with Python 2.7](https://docs.anaconda.com/anaconda/install/)

2. Installed Phyluce (used extra advice from [here](https://github.com/faircloth-lab/phyluce/issues/57) to create a new environment)
```{bash}
M053-303-26932:~ cran5048$ conda create --name phyluce
Solving environment: done


## Package Plan ##

  environment location: /anaconda2/envs/phyluce


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use:
# > source activate phyluce
#
# To deactivate an active environment, use:
# > source deactivate
#
M053-303-26932:~ cran5048$ source activate phyluce
(phyluce) M053-303-26932:~ cran5048$ conda install phyluce
Solving environment: done


M053-303-26932:~ cran5048$ conda install phyluce
Downloading and Extracting Packages
gblocks 0.91b: ########################################################################## | 100% 
raxml 8.0.19: ########################################################################### | 100% 
abyss 1.3.7: ############################################################################ | 100% 
pandas 0.13.0: ########################################################################## | 100% 
dateutil 2.4.1: ######################################################################### | 100% 
pysam 0.7.7: ############################################################################ | 100% 
bx-python 0.7.1: ######################################################################## | 100% 
biopython 1.63: ######################################################################### | 100% 
numpy 1.7.1: ############################################################################ | 100% 
mafft 7.130: ############################################################################ | 100% 
trimmomatic 0.32: ####################################################################### | 100% 
pip 9.0.3: ############################################################################## | 100% 
wheel 0.31.0: ########################################################################### | 100% 
bedtools 2.18.1: ######################################################################## | 100% 
bwa 0.7.7: ############################################################################## | 100% 
python 2.7.14: ########################################################################## | 100% 
illumiprocessor 2.0.7: ################################################################## | 100% 
samtools 0.1.19: ######################################################################## | 100% 
muscle 3.8.31: ########################################################################## | 100% 
setuptools 39.0.1: ###################################################################### | 100% 
pyvcf 0.6.4: ############################################################################ | 100% 
condaflmake 3.8.2: ###################################################################### | 100% 
dendropy 3.12.0: ######################################################################## | 100% 
bowtie 1.1.1: ########################################################################### | 100% 
trinity 2.0.6: ########################################################################## | 100% 
distribute 0.6.45: ###################################################################### | 100% 
velvet 1.2.10: ########################################################################## | 100% 
lastz 1.02.00: ########################################################################## | 100% 
phyluce 1.5.0: ########################################################################## | 100% 
picard 1.106: ########################################################################### | 100% 
gatk-lite 2.3.0: ######################################################################## | 100% 
pytz 2018.3: ############################################################################ | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done


#illumiprocessor isn't working so I removed it and reinstalled it (used force to only remove this package)
conda remove illumiprocessor --force
conda install illumiprocessor #upgrades from 2.0.7 to 2.0.8

# and install Java 8 (I have 10 installed)
brew cask install caskroom/versions/java8
#now install jenv to manage java versions, add init line to .bash_profile as suggested here https://gist.github.com/tomysmile/a9a7aee85ff73454bd57e198ad90e614
brew install jenv
#find paths to java installations
/usr/libexec/java_home -V
# add java8 to jenv
jenv add /Library/Java/JavaVirtualMachines/jdk1.8.0_162.jdk/Contents/Home
jenv versions
#set java version to 8
jenv local 1.8.0.162

#So I fixed the lines 28 and 33 in #~/anaconda2/envs/phyluce/lib/python2.7/site-packages/illumiprocessor/cli/main.py

#to add "anaconda2" in the paths there (which originally just had "anaconda")

```

3. Rename directories to reflect the sample

```{bash, evaluate=F}
for dir in */
  do
  cd $dir
  for file in *.gz; do
    cd ..
    mv $dir ${file%%_L001_R1_001.fastq.gz}
    break 1
    done
  done
  "$dir"*.gz
```

4. Copy all files into phyluce directory

```{bash, evaluate=F}
for dir in */; do
cd $dir
  for f in *.gz; do
  cp $f ../phyluce/$f
  done
cd ..
done

```

5. Cat together two individuals that were sequenced twice (once when I thought I wasn't going to include most S. ovalis in the run)

```{bash, evaluate=F}
cat sebova-sbc-R010965_S23_L001_R1_001.fastq.gz sebova-sbc-R010965_S44_L001_R1_001.fastq.gz > sebova-sbc-R010965_S2344_L001_R1_001.fastq.gz


cat sebova-sbc-R010965_S23_L001_R2_001.fastq.gz sebova-sbc-R010965_S44_L001_R2_001.fastq.gz > sebova-sbc-R010965_S2344_L001_R2_001.fastq.gz


cat sebova-sbc-R010966_S24_L001_R1_001.fastq.gz sebova-sbc-R010966_S45_L001_R1_001.fastq.gz > sebova-sbc-R010966_S2445_L001_R1_001.fastq.gz


cat sebova-sbc-R010966_S24_L001_R2_001.fastq.gz sebova-sbc-R010966_S45_L001_R2_001.fastq.gz > sebova-sbc-R010966_S2445_L001_R2_001.fastq.gz


rm sebova-sbc-R010965_S23_L001_R1_001.fastq.gz sebova-sbc-R010965_S44_L001_R1_001.fastq.gz
rm sebova-sbc-R010965_S23_L001_R2_001.fastq.gz sebova-sbc-R010965_S44_L001_R2_001.fastq.gz
rm sebova-sbc-R010966_S24_L001_R1_001.fastq.gz sebova-sbc-R010966_S45_L001_R1_001.fastq.gz
rm sebova-sbc-R010966_S24_L001_R2_001.fastq.gz sebova-sbc-R010966_S45_L001_R2_001.fastq.gz
```

# Illumiprocessor

4. Create configuration file (illumiprocessor_UCE2_config.txt)
5. Run [illumiprocessor](http://illumiprocessor.readthedocs.io/en/latest/usage.html) to trim adapters and low quality bases
```{bash, evaluate=F}
illumiprocessor --input ./ --output uce-clean --config illumiprocessor_UCE2_config.txt --cores 4

2018-04-04 18:19:59,968 - illumiprocessor - INFO - ==================== Starting illumiprocessor ===================
2018-04-04 18:19:59,990 - illumiprocessor - INFO - Version: 2.0.6
2018-04-04 18:19:59,990 - illumiprocessor - INFO - Argument --config: illumiprocessor_UCE2_config.txt
2018-04-04 18:19:59,990 - illumiprocessor - INFO - Argument --cores: 4
2018-04-04 18:19:59,990 - illumiprocessor - INFO - Argument --input: /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce
2018-04-04 18:19:59,991 - illumiprocessor - INFO - Argument --log_path: None
2018-04-04 18:19:59,991 - illumiprocessor - INFO - Argument --min_len: 40
2018-04-04 18:19:59,991 - illumiprocessor - INFO - Argument --no_merge: False
2018-04-04 18:19:59,991 - illumiprocessor - INFO - Argument --output: /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/uce-clean
2018-04-04 18:19:59,991 - illumiprocessor - INFO - Argument --phred: phred33
2018-04-04 18:19:59,991 - illumiprocessor - INFO - Argument --r1_pattern: None
2018-04-04 18:19:59,991 - illumiprocessor - INFO - Argument --r2_pattern: None
2018-04-04 18:19:59,991 - illumiprocessor - INFO - Argument --se: False
2018-04-04 18:19:59,991 - illumiprocessor - INFO - Argument --trimmomatic: /Users/cran5048/anaconda2/envs/phyluce/jar/trimmomatic.jar
2018-04-04 18:19:59,991 - illumiprocessor - INFO - Argument --verbosity: INFO
2018-04-04 18:20:00,410 - illumiprocessor - INFO - Trimming samples with Trimmomatic
Running......................................................................
2018-04-04 18:26:09,219 - illumiprocessor - INFO - =================== Completed illumiprocessor ===================

```

# Assembly

6. Create configuration file (assembly_conf.txt)
  - make this using the sample sheet in Excel to creat the paths. Need to use absolute paths to each directory
7. Run velvet
```{bash, evaluate=F}
mkdir assembly
# make a directory for log files
mkdir log
# run the assembly
phyluce_assembly_assemblo_velvet \
    --config assembly_conf.txt \
    --output ~/Basespace/UCE_Fish_2/phyluce/assembly \
    --kmer 35 \
    --subfolder split-adapter-quality-trimmed \
    --cores 4 \
    --clean \
    --log-path log

```
This took 3-4 hours to run
# Finding UCEs
8. Use LastZ to find matches to UCE probes among contigs - place them in an sql database
```{bash, evaluate=F}
mkdir FindUCE_log
# match contigs to probes
phyluce_assembly_match_contigs_to_probes \
    --contigs ./assembly/contigs \
    --probes fish-uce-500-probes.fasta \
    --output ./findUCE \
    --log-path FindUCE_log
```
This ran pretty quickly - 30 mins?

9. Check out the database
```{bash, evaluate=F}
(phyluce) M053-303-26932:FindUCE cran5048$ sqlite3 probe.matches.sqlite
SQLite version 3.22.0 2018-01-22 18:45:57
Enter ".help" for usage hints.
```

```{r}
library(DBI)
db<- dbConnect(RSQLite::SQLite(),dbname="~/Basespace/UCE_Fish_2/phyluce/FindUCE/probe.matches.sqlite")

```

```{sql connection=db, output.var="allUCEs"}

SELECT * FROM matches;




```

```{r}
rownames(allUCEs)<-allUCEs$uce

sebgoo<-allUCEs[,grep("sebgoo",names(allUCEs))]

sebova<-allUCEs[,grep("sebova",names(allUCEs))]

emblat<-allUCEs[,grep("emblat",names(allUCEs))]

embjac<-allUCEs[,grep("embjac",names(allUCEs))]

which(complete.cases(sebgoo))
which(complete.cases(sebova))
which(complete.cases(embjac))
which(complete.cases(emblat))

sebgoo<-apply(sebgoo[,1:20], FUN=as.numeric,MARGIN=2)
sebgoo_sum<-rowSums(sebgoo,na.rm=T)
names(sebgoo_sum)<-rownames(allUCEs)
sort(sebgoo_sum)

sebova<-apply(sebova[,1:17], FUN=as.numeric,MARGIN=2)
sebova_sum<-rowSums(sebova,na.rm=T)
names(sebova_sum)<-rownames(allUCEs)
sort(sebova_sum)

embjac<-apply(embjac[,1:17], FUN=as.numeric,MARGIN=2)
embjac_sum<-rowSums(embjac,na.rm=T)
names(embjac_sum)<-rownames(allUCEs)
sort(embjac_sum)

emblat<-apply(emblat[,1:15], FUN=as.numeric,MARGIN=2)
emblat_sum<-rowSums(emblat,na.rm=T)
names(emblat_sum)<-rownames(allUCEs)
sort(emblat_sum)

```
So there were `length(which(complete.cases(allUCEs)))` UCEs that got enriched in all samples :-( but not surprising as there were 72 taxa.

So I might want to back up and have it do databases for each species. Or actually I can subselect taxon sets below. But I will plow ahead and use all data for now

10. Make a list of all taxa and UCEs that I can get for the whole dataset (this is just a test prior to running the next steps)
```{bash, evaluate=F}
phyluce_assembly_get_match_counts \
    --locus-db /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/FindUCE/probe.matches.sqlite \
    --taxon-list-config /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/all_taxa_sets_conf.txt\
    --taxon-group 'all_taxa_dataset' \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa/alltaxa_dataset1_conf.txt \
    --incomplete-matrix
    
```

11. Extract all the matching data into a FASTA
```{bash, evaluate=F}
phyluce_assembly_get_fastas_from_match_counts \
    --contigs /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/assembly/contigs/ \
    --locus-db /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/FindUCE/probe.matches.sqlite \
    --match-count-output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa/alltaxa_dataset1_conf.txt \
    --incomplete-matrix /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa/alltaxa_dataset1.incomplete \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa/alltaxa_dataset1.fasta
    
```

12. Align the FASTA Reads
```{bash, evaluate=F}

phyluce_align_seqcap_align \
    --fasta /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa/alltaxa_dataset1.fasta \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa/alignments \
    --taxa 70 \
    --aligner mafft \
    --cores 4 \
    --proportion 0.05 \
    --threshold 0.05 \
    --incomplete-matrix \
    --max-divergence 0.05 \
    --min-length 50 \
```

After playing with the data I set a minimum length of 50, a proportion of 0.1 and max divergence of 0.05 and found 3 loci with intraspecific variation
51 (sebova), 1169 (sebgoo, sebova), 1296-(sebgoo)

# Backing up to Assembly

7. Run velvet with larger kmer value
```{bash, evaluate=F}
mkdir assembly
# make a directory for log files
mkdir log
# run the assembly
phyluce_assembly_assemblo_velvet \
    --config assembly_conf.txt \
    --output ~/Basespace/UCE_Fish_2/phyluce/assembly \
    --kmer 65 \
    --subfolder split-adapter-quality-trimmed \
    --cores 4 \
    --clean \
    --log-path log

```

8. Use LastZ to find matches to UCE probes among contigs - place them in an sql database
```{bash, evaluate=F}
mkdir FindUCE_log
# match contigs to probes
phyluce_assembly_match_contigs_to_probes \
    --contigs ./assembly/contigs \
    --probes fish-uce-500-probes.fasta \
    --output ./findUCE_k65 \
    --log-path FindUCE_log_k65
```


10. Make a list of all taxa and UCEs that I can get for the whole dataset (this is just a test prior to running the next steps)
```{bash, evaluate=F}
phyluce_assembly_get_match_counts \
    --locus-db /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/FindUCE_k65/probe.matches.sqlite \
    --taxon-list-config /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/all_taxa_sets_conf.txt\
    --taxon-group 'all_taxa_dataset' \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa_k65/alltaxa_dataset1_conf.txt \
    --incomplete-matrix
    
```

11. Extract all the matching data into a FASTA
```{bash, evaluate=F}
phyluce_assembly_get_fastas_from_match_counts \
    --contigs /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/assembly_k65/contigs/ \
    --locus-db /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/FindUCE_k65/probe.matches.sqlite \
    --match-count-output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa_k65/alltaxa_dataset1_conf.txt \
    --incomplete-matrix /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa_k65/alltaxa_dataset1.incomplete \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa_k65/alltaxa_dataset1.fasta
    
```


12. Align the FASTA Reads
```{bash, evaluate=F}

phyluce_align_seqcap_align \
    --fasta /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa_k65/alltaxa_dataset1.fasta \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/alltaxa_k65/alignments \
    --taxa 70 \
    --aligner mafft \
    --cores 4 \
    --proportion 0.75 \
    --incomplete-matrix \
    --max-divergence 0.05 \
    --min-length 50 \
```

UCE-1020 UCE34 UCE103 UCE109 (Seb) UCE129 (Seb) UCE167 (Seb) UCE245 (Seb) UCE267 (Seb) UCE 307 (Seb) UCE 344( Sebova) UCE 418 (sebova) UCE 431 (Sebgoo) UCE 461 (Sebgoo) UCE377 (Sebova)


# Breaking it down by species
10. Make a list of all taxa and UCEs that I can get for the whole dataset (this is just a test prior to running the next steps)
```{bash, evaluate=F}
phyluce_assembly_get_match_counts \
    --locus-db /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/FindUCE_k65/probe.matches.sqlite \
    --taxon-list-config /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/all_taxa_sets_conf.txt\
    --taxon-group 'sebgoo' \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebgoo/sebgoo_dataset1_conf.txt \
    --incomplete-matrix
    
    
    phyluce_assembly_get_match_counts \
    --locus-db /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/FindUCE_k65/probe.matches.sqlite \
    --taxon-list-config /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/all_taxa_sets_conf.txt\
    --taxon-group 'sebova' \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebova/sebova_dataset1_conf.txt \
    --incomplete-matrix
    
    phyluce_assembly_get_match_counts \
    --locus-db /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/FindUCE_k65/probe.matches.sqlite \
    --taxon-list-config /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/all_taxa_sets_conf.txt\
    --taxon-group 'emblat' \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/emblat/emblat_dataset1_conf.txt \
    --incomplete-matrix
    
    phyluce_assembly_get_match_counts \
    --locus-db /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/FindUCE_k65/probe.matches.sqlite \
    --taxon-list-config /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/all_taxa_sets_conf.txt\
    --taxon-group 'embjac' \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/embjac/embjac_dataset1_conf.txt \
    --incomplete-matrix
    
```

## Sebastes Ovalis
11. Extract all the matching data into a FASTA
```{bash, evaluate=F}
phyluce_assembly_get_fastas_from_match_counts \
    --contigs /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/assembly_k65/contigs/ \
    --locus-db /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/FindUCE_k65/probe.matches.sqlite \
    --match-count-output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebova/sebova_dataset1_conf.txt \
    --incomplete-matrix /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebova/sebova_dataset1.incomplete \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebova/sebova_dataset1.fasta
    
```

Align the FASTA Reads
```{bash, evaluate=F}

phyluce_align_seqcap_align \
    --fasta /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebova/sebova_dataset1.fasta \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebova/alignments \
    --taxa 17 \
    --aligner mafft \
    --cores 4 \
    --proportion 0.75 \
    --incomplete-matrix \
    --max-divergence 0.03 \
    --min-length 50
```



Get summary stats
```{bash, evaluate=F}
phyluce_align_get_align_summary_data \
    --alignments ./sebova/alignments \
    --cores 4 \
    --log-path log
    
    2018-04-07 14:29:57,493 - phyluce_align_get_align_summary_data - INFO - ========= Starting phyluce_align_get_align_summary_data =========
2018-04-07 14:29:57,493 - phyluce_align_get_align_summary_data - INFO - Version: 1.5.0
2018-04-07 14:29:57,493 - phyluce_align_get_align_summary_data - INFO - Argument --alignments: /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebova/alignments
2018-04-07 14:29:57,493 - phyluce_align_get_align_summary_data - INFO - Argument --cores: 4
2018-04-07 14:29:57,494 - phyluce_align_get_align_summary_data - INFO - Argument --input_format: nexus
2018-04-07 14:29:57,494 - phyluce_align_get_align_summary_data - INFO - Argument --log_path: /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/log
2018-04-07 14:29:57,494 - phyluce_align_get_align_summary_data - INFO - Argument --show_taxon_counts: False
2018-04-07 14:29:57,494 - phyluce_align_get_align_summary_data - INFO - Argument --verbosity: INFO
2018-04-07 14:29:57,494 - phyluce_align_get_align_summary_data - INFO - Getting alignment files
2018-04-07 14:29:57,565 - phyluce_align_get_align_summary_data - INFO - Computing summary statistics using 4 cores
2018-04-07 14:29:58,807 - phyluce_align_get_align_summary_data - INFO - ----------------------- Alignment summary -----------------------
2018-04-07 14:29:58,807 - phyluce_align_get_align_summary_data - INFO - [Alignments] loci:	477
2018-04-07 14:29:58,807 - phyluce_align_get_align_summary_data - INFO - [Alignments] length:	79,651
2018-04-07 14:29:58,807 - phyluce_align_get_align_summary_data - INFO - [Alignments] mean:	166.98
2018-04-07 14:29:58,807 - phyluce_align_get_align_summary_data - INFO - [Alignments] 95% CI:	5.94
2018-04-07 14:29:58,807 - phyluce_align_get_align_summary_data - INFO - [Alignments] min:	55
2018-04-07 14:29:58,807 - phyluce_align_get_align_summary_data - INFO - [Alignments] max:	484
2018-04-07 14:29:58,808 - phyluce_align_get_align_summary_data - INFO - ------------------------- Taxon summary -------------------------
2018-04-07 14:29:58,808 - phyluce_align_get_align_summary_data - INFO - [Taxa] mean:		7.87
2018-04-07 14:29:58,808 - phyluce_align_get_align_summary_data - INFO - [Taxa] 95% CI:	0.21
2018-04-07 14:29:58,808 - phyluce_align_get_align_summary_data - INFO - [Taxa] min:		3
2018-04-07 14:29:58,808 - phyluce_align_get_align_summary_data - INFO - [Taxa] max:		15
2018-04-07 14:29:58,809 - phyluce_align_get_align_summary_data - INFO - ----------------- Missing data from trim summary ----------------
2018-04-07 14:29:58,809 - phyluce_align_get_align_summary_data - INFO - [Missing] mean:	10.06
2018-04-07 14:29:58,809 - phyluce_align_get_align_summary_data - INFO - [Missing] 95% CI:	0.36
2018-04-07 14:29:58,809 - phyluce_align_get_align_summary_data - INFO - [Missing] min:	0.84
2018-04-07 14:29:58,809 - phyluce_align_get_align_summary_data - INFO - [Missing] max:	22.46
2018-04-07 14:29:58,814 - phyluce_align_get_align_summary_data - INFO - -------------------- Character count summary --------------------
2018-04-07 14:29:58,814 - phyluce_align_get_align_summary_data - INFO - [All characters]	613,544
2018-04-07 14:29:58,814 - phyluce_align_get_align_summary_data - INFO - [Nucleotides]		546,272
2018-04-07 14:29:58,814 - phyluce_align_get_align_summary_data - INFO - ---------------- Data matrix completeness summary ---------------
2018-04-07 14:29:58,815 - phyluce_align_get_align_summary_data - INFO - [Matrix 50%]		267 alignments
2018-04-07 14:29:58,815 - phyluce_align_get_align_summary_data - INFO - [Matrix 55%]		180 alignments
2018-04-07 14:29:58,815 - phyluce_align_get_align_summary_data - INFO - [Matrix 60%]		180 alignments
2018-04-07 14:29:58,815 - phyluce_align_get_align_summary_data - INFO - [Matrix 65%]		119 alignments
2018-04-07 14:29:58,815 - phyluce_align_get_align_summary_data - INFO - [Matrix 70%]		68 alignments
2018-04-07 14:29:58,815 - phyluce_align_get_align_summary_data - INFO - [Matrix 75%]		29 alignments
2018-04-07 14:29:58,815 - phyluce_align_get_align_summary_data - INFO - [Matrix 80%]		29 alignments
2018-04-07 14:29:58,815 - phyluce_align_get_align_summary_data - INFO - [Matrix 85%]		9 alignments
2018-04-07 14:29:58,815 - phyluce_align_get_align_summary_data - INFO - [Matrix 90%]		5 alignments
2018-04-07 14:29:58,816 - phyluce_align_get_align_summary_data - INFO - [Matrix 95%]		1 alignments
2018-04-07 14:29:58,816 - phyluce_align_get_align_summary_data - INFO - ------------------------ Character counts -----------------------
2018-04-07 14:29:58,816 - phyluce_align_get_align_summary_data - INFO - [Characters] '-' is present 2,617 times
2018-04-07 14:29:58,816 - phyluce_align_get_align_summary_data - INFO - [Characters] '?' is present 64,655 times
2018-04-07 14:29:58,816 - phyluce_align_get_align_summary_data - INFO - [Characters] 'A' is present 148,350 times
2018-04-07 14:29:58,816 - phyluce_align_get_align_summary_data - INFO - [Characters] 'C' is present 124,944 times
2018-04-07 14:29:58,816 - phyluce_align_get_align_summary_data - INFO - [Characters] 'G' is present 124,453 times
2018-04-07 14:29:58,816 - phyluce_align_get_align_summary_data - INFO - [Characters] 'T' is present 148,525 times
2018-04-07 14:29:58,817 - phyluce_align_get_align_summary_data - INFO - ========= Completed phyluce_align_get_align_summary_data ========


```

```{bash, evaluate=F}
phyluce_align_explode_alignments \
    --alignments ./sebova/alignments \
    --output ./sebova/exploded-fastas \
    --input-format nexus \
    --by-taxon
    
```

Create a phasing config file as described [here](http://phyluce.readthedocs.io/en/latest/tutorial-two.html)

`phyluce_snp_bwa_multiple_align` was not included in the distribution. I copied it from [here](https://github.com/faircloth-lab/phyluce/blob/master/bin/snps/phyluce_snp_bwa_multiple_align)
```{bash, evaluate=F}
phyluce_snp_bwa_multiple_align \
    --config ./sebova/sebova_phasing_conf.txt \
    --output ./sebova/mapping \
    --subfolder split-adapter-quality-trimmed \
    --log-path ./sebova/bwa_log \
    --cores 4
    
```

To get this to work, I had to clone the latest git version of phyluce into my github directory and add
it to my python path.
```{bash, evaluate=F}
export PYTHONPATH=$HOME/github/phyluce

python ~/github/phyluce/bin/snps/phyluce_snp_phase_uces \
    --config ./sebova/sebova_phasing_conf.txt \
    --bams ./sebova/mapping \
    --output ./sebova/phased_reads
```

### Samtools
Going off on my own now, trying to call SNPs in the mapped bam files. Following guidance from [here](http://samtools.sourceforge.net/mpileup.shtml).

```{bash, evaluate=F}
(phyluce) M053-303-26932:mapping cran5048$ pwd
/Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebova/mapping
(phyluce) M053-303-26932:mapping cran5048$ find . -name "*.bam" -exec cp {} ./bams \;
(phyluce) M053-303-26932:mapping cran5048$ find . -name "*.bai" -exec cp {} ./bams \;

#Following guidance from 
#index the fasta file at the top of the sebova directory
(phyluce) M053-303-26932:sebova cran5048$ pwd
/Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebova
(phyluce) M053-303-26932:sebova cran5048$ samtools faidx
Usage: faidx <in.fasta> [<reg> [...]]
(phyluce) M053-303-26932:sebova cran5048$ samtools faidx 
.DS_Store                   exploded-fastas/            sebova_dataset1.fasta       sebova_phasing_conf.txt
alignments/                 mapping/                    sebova_dataset1.incomplete  
bwa_log/                    phased_reads/               sebova_dataset1_conf.txt    
(phyluce) M053-303-26932:sebova cran5048$ samtools faidx sebova_dataset1.fasta

#use mpileup to compute likelihood of each genotype
(phyluce) M053-303-26932:sebova cran5048$ cd mapping
(phyluce) M053-303-26932:mapping cran5048$ cd bams
(phyluce) M053-303-26932:bams cran5048$ samtools mpileup -uf ../../sebova_dataset1.fasta *.bam | bcftools view -bvcg - > var.raw3.bcf
[mpileup] 17 samples in 17 input files
<mpileup> Set max per-file depth to 470
[afs] 0:6361.738 1:634.648 2:644.429 3:656.019 4:667.827 5:679.541 6:691.061 7:702.351 8:713.395 9:724.189 10:734.735 11:745.037 12:755.103 13:764.940 14:774.556 15:783.963 16:793.170 17:802.188 18:811.029 19:819.704 20:828.226 21:836.607 22:844.859 23:852.995 24:861.028 25:868.971 26:876.836 27:884.638 28:892.388 29:900.100 30:907.787 31:915.461 32:923.134 33:930.819 34:938.527

#call the SNPs
(phyluce) M053-303-26932:bams cran5048$ bcftools view var.raw3.bcf | vcfutils.pl varFilter -d5  -D100 > var.flt3.vcf

````

This wasn't working - it was calling a SNP for virtually every position of the alignment. I think I am using an incorrect reference fasta for SNP calling. The .bam files were aligned against sample-specific references, so that is technically the reference that the SNPs need to be called against. Instead, I will use this other `bwa` alignment script that I found in phyluce to align against a single reference, that I created by loading all the nexus files from the trimming step into Geneious and exporting consensus sequences.

```{bash, evaluate=F}
phyluce_snp_bwa_align \
    --config ./sebova/sebova_phasing_conf2.txt \
    --output ./sebova/mapping2 \
    --subfolder split-adapter-quality-trimmed \
    --log-path ./sebova/bwa_log \
    --cores 4
    
    
    
(phyluce) M053-303-26932:mapping2 cran5048$ mkdir bams
(phyluce) M053-303-26932:mapping2 cran5048$ find . -name "*.bam" -exec cp {} ./bams \;
(phyluce) M053-303-26932:mapping2 cran5048$ cd bams

for bam in *.bam
do
samtools index $bam
done

# mapping didn't work on R010970 for some reason

samtools faidx ../../contigs_consensus.fasta

(phyluce) M053-303-26932:bams cran5048$ samtools mpileup -uf ../../contigs_consensus.fasta *.bam | bcftools view -bvcgI  - > var.raw4.bcf
[mpileup] 17 samples in 17 input files
<mpileup> Set max per-file depth to 470
[afs] 0:1401.204 1:36.449 2:24.606 3:16.974 4:12.449 5:9.728 6:7.988 7:6.800 8:5.942 9:5.289 10:4.767 11:4.334 12:3.971 13:3.672 14:3.424 15:3.209 16:3.007 17:2.797 18:2.575 19:2.346 20:2.124 21:1.923 22:1.751 23:1.609 24:1.497 25:1.412 26:1.354 27:1.326 28:1.333 29:1.385 30:1.501 31:1.721 32:2.134 33:2.923 34:4.475


 bcftools view var.raw4.bcf | vcfutils.pl varFilter -d5 -Q30 -D100 > var.flt4.vcf
 
 
(phyluce) M053-303-26932:bams cran5048$ phyluce_snp_convert_vcf_to_structure --input var.flt4.vcf --output sebova_structure.txt
Working.
Data contain 17 individuals and 91 loci output to sebova_structure.txt.
 
```

This created a vcf file with credible SNPs! Although many of the SNPs seem to have an excess of heterozygotes...

```{r}
library(pegas)
library(adegenet)



#sebova<-read.vcf("~/BaseSpace/UCE_Fish_2/phyluce/sebova/mapping2/bams/var.flt4.vcf")
  sebova<-read.genepop("~/BaseSpace/UCE_Fish_2/phyluce/sebova/mapping2/bams/sebova.gen",ncode = 3)

```


### GATK

```{bash, evaluate=F}

gatk  HaplotypeCaller \
      -R ../../contigs_consensus.fasta \
      -I sebova_cordell_R010969-CL-RG-MD-M.bam \
      -O test.g.vcf \
```



## Sebastes goodei
11. Extract all the matching data into a FASTA
```{bash, evaluate=F}
phyluce_assembly_get_fastas_from_match_counts \
    --contigs /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/assembly_k65/contigs/ \
    --locus-db /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/FindUCE_k65/probe.matches.sqlite \
    --match-count-output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebgoo/sebgoo_dataset1_conf.txt \
    --incomplete-matrix /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebgoo/sebgoo_dataset1.incomplete\
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebgoo/sebgoo_dataset1.fasta
    
```

Align the FASTA Reads
```{bash, evaluate=F}

phyluce_align_seqcap_align \
    --fasta /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebgoo/sebgoo_dataset1.fasta \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebgoo/alignments \
    --taxa 17 \
    --aligner mafft \
    --cores 4 \
    --proportion 0.75 \
    --incomplete-matrix \
    --max-divergence 0.03 \
    --min-length 50
    
[WARNING] Output directory exists, REMOVE [Y/n]? Y
2018-04-24 13:49:11,303 - phyluce_align_seqcap_align - INFO - ============== Starting phyluce_align_seqcap_align ==============
2018-04-24 13:49:11,320 - phyluce_align_seqcap_align - INFO - Version: 1.5.0
2018-04-24 13:49:11,320 - phyluce_align_seqcap_align - INFO - Argument --aligner: mafft
2018-04-24 13:49:11,320 - phyluce_align_seqcap_align - INFO - Argument --ambiguous: False
2018-04-24 13:49:11,320 - phyluce_align_seqcap_align - INFO - Argument --cores: 4
2018-04-24 13:49:11,320 - phyluce_align_seqcap_align - INFO - Argument --fasta: /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebgoo/sebgoo_dataset1.fasta
2018-04-24 13:49:11,321 - phyluce_align_seqcap_align - INFO - Argument --log_path: None
2018-04-24 13:49:11,321 - phyluce_align_seqcap_align - INFO - Argument --max_divergence: 0.03
2018-04-24 13:49:11,321 - phyluce_align_seqcap_align - INFO - Argument --min_length: 50
2018-04-24 13:49:11,321 - phyluce_align_seqcap_align - INFO - Argument --no_trim: False
2018-04-24 13:49:11,321 - phyluce_align_seqcap_align - INFO - Argument --notstrict: True
2018-04-24 13:49:11,321 - phyluce_align_seqcap_align - INFO - Argument --output: /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebgoo/alignments
2018-04-24 13:49:11,321 - phyluce_align_seqcap_align - INFO - Argument --output_format: nexus
2018-04-24 13:49:11,321 - phyluce_align_seqcap_align - INFO - Argument --proportion: 0.75
2018-04-24 13:49:11,321 - phyluce_align_seqcap_align - INFO - Argument --taxa: 17
2018-04-24 13:49:11,321 - phyluce_align_seqcap_align - INFO - Argument --threshold: 0.65
2018-04-24 13:49:11,322 - phyluce_align_seqcap_align - INFO - Argument --verbosity: INFO
2018-04-24 13:49:11,322 - phyluce_align_seqcap_align - INFO - Argument --window: 20
2018-04-24 13:49:11,322 - phyluce_align_seqcap_align - INFO - Building the locus dictionary
2018-04-24 13:49:11,322 - phyluce_align_seqcap_align - INFO - Removing ALL sequences with ambiguous bases...
2018-04-24 13:49:11,810 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-136. Too few taxa (N < 3).
2018-04-24 13:49:11,810 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-1048. Too few taxa (N < 3).
2018-04-24 13:49:11,810 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-63. Too few taxa (N < 3).
2018-04-24 13:49:11,813 - phyluce_align_seqcap_align - INFO - Aligning with MAFFT
2018-04-24 13:49:11,814 - phyluce_align_seqcap_align - INFO - Alignment begins. 'X' indicates dropped alignments (these are reported after alignment)
...........X...........................................X..X.......................................X...X........X...............X.........X......X...X.........................................................................................X.........................................X...................X.......X..............................................................X.....................X...........X...................X..................................X........................X..........
2018-04-24 13:49:55,942 - phyluce_align_seqcap_align - INFO - Alignment ends
2018-04-24 13:49:55,942 - phyluce_align_seqcap_align - INFO - Writing output files
2018-04-24 13:49:55,949 - phyluce_align_seqcap_align - WARNING - DROPPED uce-374 from output
2018-04-24 13:49:55,961 - phyluce_align_seqcap_align - WARNING - DROPPED uce-76 from output
2018-04-24 13:49:55,967 - phyluce_align_seqcap_align - WARNING - DROPPED uce-1147 from output
2018-04-24 13:49:55,992 - phyluce_align_seqcap_align - WARNING - DROPPED uce-685 from output
2018-04-24 13:49:55,992 - phyluce_align_seqcap_align - WARNING - DROPPED uce-1188 from output
2018-04-24 13:49:56,012 - phyluce_align_seqcap_align - WARNING - DROPPED uce-1115 from output
2018-04-24 13:49:56,015 - phyluce_align_seqcap_align - WARNING - DROPPED uce-612 from output
2018-04-24 13:49:56,030 - phyluce_align_seqcap_align - WARNING - DROPPED uce-369 from output
2018-04-24 13:49:56,050 - phyluce_align_seqcap_align - WARNING - DROPPED uce-1012 from output
2018-04-24 13:49:56,072 - phyluce_align_seqcap_align - WARNING - DROPPED uce-256 from output
2018-04-24 13:49:56,072 - phyluce_align_seqcap_align - WARNING - DROPPED uce-319 from output
2018-04-24 13:49:56,115 - phyluce_align_seqcap_align - WARNING - DROPPED uce-412 from output
2018-04-24 13:49:56,147 - phyluce_align_seqcap_align - WARNING - DROPPED uce-658 from output
2018-04-24 13:49:56,150 - phyluce_align_seqcap_align - WARNING - DROPPED uce-514 from output
2018-04-24 13:49:56,193 - phyluce_align_seqcap_align - WARNING - DROPPED uce-639 from output
2018-04-24 13:49:56,194 - phyluce_align_seqcap_align - WARNING - DROPPED uce-733 from output
2018-04-24 13:49:56,223 - phyluce_align_seqcap_align - WARNING - DROPPED uce-231 from output
2018-04-24 13:49:56,226 - phyluce_align_seqcap_align - WARNING - DROPPED uce-541 from output
2018-04-24 13:49:56,232 - phyluce_align_seqcap_align - WARNING - DROPPED uce-1253 from output
2018-04-24 13:49:56,245 - phyluce_align_seqcap_align - WARNING - DROPPED uce-1087 from output
2018-04-24 13:49:56,265 - phyluce_align_seqcap_align - INFO - ============== Completed phyluce_align_seqcap_align =============

```



Get summary stats
```{bash, evaluate=F}
phyluce_align_get_align_summary_data \
    --alignments ./sebgoo/alignments \
    --cores 4 \
    --log-path log
    
    
2018-04-24 13:52:31,863 - phyluce_align_get_align_summary_data - INFO - ========= Starting phyluce_align_get_align_summary_data =========
2018-04-24 13:52:31,882 - phyluce_align_get_align_summary_data - INFO - Version: 1.5.0
2018-04-24 13:52:31,882 - phyluce_align_get_align_summary_data - INFO - Argument --alignments: /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/sebgoo/alignments
2018-04-24 13:52:31,882 - phyluce_align_get_align_summary_data - INFO - Argument --cores: 4
2018-04-24 13:52:31,882 - phyluce_align_get_align_summary_data - INFO - Argument --input_format: nexus
2018-04-24 13:52:31,882 - phyluce_align_get_align_summary_data - INFO - Argument --log_path: /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/log
2018-04-24 13:52:31,882 - phyluce_align_get_align_summary_data - INFO - Argument --show_taxon_counts: False
2018-04-24 13:52:31,882 - phyluce_align_get_align_summary_data - INFO - Argument --verbosity: INFO
2018-04-24 13:52:31,882 - phyluce_align_get_align_summary_data - INFO - Getting alignment files
2018-04-24 13:52:31,885 - phyluce_align_get_align_summary_data - INFO - Computing summary statistics using 4 cores
2018-04-24 13:52:32,146 - phyluce_align_get_align_summary_data - INFO - ----------------------- Alignment summary -----------------------
2018-04-24 13:52:32,147 - phyluce_align_get_align_summary_data - INFO - [Alignments] loci:	476
2018-04-24 13:52:32,147 - phyluce_align_get_align_summary_data - INFO - [Alignments] length:	57,562
2018-04-24 13:52:32,147 - phyluce_align_get_align_summary_data - INFO - [Alignments] mean:	120.93
2018-04-24 13:52:32,147 - phyluce_align_get_align_summary_data - INFO - [Alignments] 95% CI:	3.17
2018-04-24 13:52:32,147 - phyluce_align_get_align_summary_data - INFO - [Alignments] min:	51
2018-04-24 13:52:32,147 - phyluce_align_get_align_summary_data - INFO - [Alignments] max:	369
2018-04-24 13:52:32,148 - phyluce_align_get_align_summary_data - INFO - ------------------------- Taxon summary -------------------------
2018-04-24 13:52:32,148 - phyluce_align_get_align_summary_data - INFO - [Taxa] mean:		8.32
2018-04-24 13:52:32,148 - phyluce_align_get_align_summary_data - INFO - [Taxa] 95% CI:	0.20
2018-04-24 13:52:32,148 - phyluce_align_get_align_summary_data - INFO - [Taxa] min:		3
2018-04-24 13:52:32,148 - phyluce_align_get_align_summary_data - INFO - [Taxa] max:		17
2018-04-24 13:52:32,148 - phyluce_align_get_align_summary_data - INFO - ----------------- Missing data from trim summary ----------------
2018-04-24 13:52:32,149 - phyluce_align_get_align_summary_data - INFO - [Missing] mean:	9.69
2018-04-24 13:52:32,149 - phyluce_align_get_align_summary_data - INFO - [Missing] 95% CI:	0.35
2018-04-24 13:52:32,149 - phyluce_align_get_align_summary_data - INFO - [Missing] min:	0.00
2018-04-24 13:52:32,149 - phyluce_align_get_align_summary_data - INFO - [Missing] max:	27.60
2018-04-24 13:52:32,154 - phyluce_align_get_align_summary_data - INFO - -------------------- Character count summary --------------------
2018-04-24 13:52:32,154 - phyluce_align_get_align_summary_data - INFO - [All characters]	468,404
2018-04-24 13:52:32,154 - phyluce_align_get_align_summary_data - INFO - [Nucleotides]		416,553
2018-04-24 13:52:32,155 - phyluce_align_get_align_summary_data - INFO - ---------------- Data matrix completeness summary ---------------
2018-04-24 13:52:32,155 - phyluce_align_get_align_summary_data - INFO - [Matrix 50%]		219 alignments
2018-04-24 13:52:32,155 - phyluce_align_get_align_summary_data - INFO - [Matrix 55%]		149 alignments
2018-04-24 13:52:32,155 - phyluce_align_get_align_summary_data - INFO - [Matrix 60%]		81 alignments
2018-04-24 13:52:32,155 - phyluce_align_get_align_summary_data - INFO - [Matrix 65%]		81 alignments
2018-04-24 13:52:32,155 - phyluce_align_get_align_summary_data - INFO - [Matrix 70%]		35 alignments
2018-04-24 13:52:32,156 - phyluce_align_get_align_summary_data - INFO - [Matrix 75%]		13 alignments
2018-04-24 13:52:32,156 - phyluce_align_get_align_summary_data - INFO - [Matrix 80%]		5 alignments
2018-04-24 13:52:32,156 - phyluce_align_get_align_summary_data - INFO - [Matrix 85%]		4 alignments
2018-04-24 13:52:32,156 - phyluce_align_get_align_summary_data - INFO - [Matrix 90%]		3 alignments
2018-04-24 13:52:32,156 - phyluce_align_get_align_summary_data - INFO - [Matrix 95%]		3 alignments
2018-04-24 13:52:32,156 - phyluce_align_get_align_summary_data - INFO - ------------------------ Character counts -----------------------
2018-04-24 13:52:32,156 - phyluce_align_get_align_summary_data - INFO - [Characters] '-' is present 5,425 times
2018-04-24 13:52:32,156 - phyluce_align_get_align_summary_data - INFO - [Characters] '?' is present 46,426 times
2018-04-24 13:52:32,156 - phyluce_align_get_align_summary_data - INFO - [Characters] 'A' is present 113,291 times
2018-04-24 13:52:32,157 - phyluce_align_get_align_summary_data - INFO - [Characters] 'C' is present 94,786 times
2018-04-24 13:52:32,157 - phyluce_align_get_align_summary_data - INFO - [Characters] 'G' is present 93,462 times
2018-04-24 13:52:32,157 - phyluce_align_get_align_summary_data - INFO - [Characters] 'T' is present 115,014 times
2018-04-24 13:52:32,157 - phyluce_align_get_align_summary_data - INFO - ========= Completed phyluce_align_get_align_summary_data ========


```

Create consensus sequences of all alignments in Geneious

Map reads to fasta
```{bash, evaluate=F}
phyluce_snp_bwa_align \
    --config ./sebgoo/sebgoo_phasing_conf.txt \
    --output ./sebgoo/mapping \
    --subfolder split-adapter-quality-trimmed \
    --log-path ./sebgoo/bwa_log \
    --cores 4
    
    
    
(phyluce) M053-303-26932:mapping2 cran5048$ mkdir bams
(phyluce) M053-303-26932:mapping2 cran5048$ find . -name "*.bam" -exec cp {} ./bams \;
(phyluce) M053-303-26932:mapping2 cran5048$ cd bams

for bam in *.bam
do
samtools index $bam
done

samtools faidx ../../Contigs_consensus.fasta

(phyluce) M053-303-26932:bams cran5048$ samtools mpileup -uf ../../contigs_consensus.fasta *.bam | bcftools view -bvcgI  - > var.raw.bcf
[mpileup] 17 samples in 17 input files
<mpileup> Set max per-file depth to 470
[afs] 0:1401.204 1:36.449 2:24.606 3:16.974 4:12.449 5:9.728 6:7.988 7:6.800 8:5.942 9:5.289 10:4.767 11:4.334 12:3.971 13:3.672 14:3.424 15:3.209 16:3.007 17:2.797 18:2.575 19:2.346 20:2.124 21:1.923 22:1.751 23:1.609 24:1.497 25:1.412 26:1.354 27:1.326 28:1.333 29:1.385 30:1.501 31:1.721 32:2.134 33:2.923 34:4.475


 bcftools view var.raw.bcf | vcfutils.pl varFilter -d5 -Q30 -D100 > var.flt4.vcf
 
 7 SNPS
```

## Embiotica jacksoni
11. Extract all the matching data into a FASTA
```{bash, evaluate=F}
phyluce_assembly_get_fastas_from_match_counts \
    --contigs /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/assembly_k65/contigs/ \
    --locus-db /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/FindUCE_k65/probe.matches.sqlite \
    --match-count-output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/embjac/embjac_dataset1_conf.txt \
    --incomplete-matrix /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/embjac/embjac_dataset1.incomplete\
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/embjac/embjac_dataset1.fasta
    
```

Align the FASTA Reads
```{bash, evaluate=F}

phyluce_align_seqcap_align \
    --fasta /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/embjac/embjac_dataset1.fasta \
    --output /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/embjac/alignments \
    --taxa 17 \
    --aligner mafft \
    --cores 4 \
    --proportion 0.75 \
    --incomplete-matrix \
    --max-divergence 0.03 \
    --min-length 50
    
2018-04-25 13:54:37,378 - phyluce_align_seqcap_align - INFO - ============== Starting phyluce_align_seqcap_align ==============
2018-04-25 13:54:37,378 - phyluce_align_seqcap_align - INFO - Version: 1.5.0
2018-04-25 13:54:37,378 - phyluce_align_seqcap_align - INFO - Argument --aligner: mafft
2018-04-25 13:54:37,378 - phyluce_align_seqcap_align - INFO - Argument --ambiguous: False
2018-04-25 13:54:37,378 - phyluce_align_seqcap_align - INFO - Argument --cores: 4
2018-04-25 13:54:37,379 - phyluce_align_seqcap_align - INFO - Argument --fasta: /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/embjac/embjac_dataset1.fasta
2018-04-25 13:54:37,379 - phyluce_align_seqcap_align - INFO - Argument --log_path: None
2018-04-25 13:54:37,379 - phyluce_align_seqcap_align - INFO - Argument --max_divergence: 0.03
2018-04-25 13:54:37,379 - phyluce_align_seqcap_align - INFO - Argument --min_length: 50
2018-04-25 13:54:37,379 - phyluce_align_seqcap_align - INFO - Argument --no_trim: False
2018-04-25 13:54:37,379 - phyluce_align_seqcap_align - INFO - Argument --notstrict: True
2018-04-25 13:54:37,379 - phyluce_align_seqcap_align - INFO - Argument --output: /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/embjac/alignments
2018-04-25 13:54:37,379 - phyluce_align_seqcap_align - INFO - Argument --output_format: nexus
2018-04-25 13:54:37,379 - phyluce_align_seqcap_align - INFO - Argument --proportion: 0.75
2018-04-25 13:54:37,379 - phyluce_align_seqcap_align - INFO - Argument --taxa: 17
2018-04-25 13:54:37,380 - phyluce_align_seqcap_align - INFO - Argument --threshold: 0.65
2018-04-25 13:54:37,380 - phyluce_align_seqcap_align - INFO - Argument --verbosity: INFO
2018-04-25 13:54:37,380 - phyluce_align_seqcap_align - INFO - Argument --window: 20
2018-04-25 13:54:37,380 - phyluce_align_seqcap_align - INFO - Building the locus dictionary
2018-04-25 13:54:37,380 - phyluce_align_seqcap_align - INFO - Removing ALL sequences with ambiguous bases...
2018-04-25 13:54:38,069 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-372. Too few taxa (N < 3).
2018-04-25 13:54:38,069 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-270. Too few taxa (N < 3).
2018-04-25 13:54:38,069 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-975. Too few taxa (N < 3).
2018-04-25 13:54:38,069 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-245. Too few taxa (N < 3).
2018-04-25 13:54:38,069 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-249. Too few taxa (N < 3).
2018-04-25 13:54:38,070 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-881. Too few taxa (N < 3).
2018-04-25 13:54:38,070 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-864. Too few taxa (N < 3).
2018-04-25 13:54:38,070 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-756. Too few taxa (N < 3).
2018-04-25 13:54:38,070 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-1238. Too few taxa (N < 3).
2018-04-25 13:54:38,070 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-408. Too few taxa (N < 3).
2018-04-25 13:54:38,070 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-4. Too few taxa (N < 3).
2018-04-25 13:54:38,070 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-695. Too few taxa (N < 3).
2018-04-25 13:54:38,070 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-1316. Too few taxa (N < 3).
2018-04-25 13:54:38,071 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-410. Too few taxa (N < 3).
2018-04-25 13:54:38,071 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-1001. Too few taxa (N < 3).
2018-04-25 13:54:38,071 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-157. Too few taxa (N < 3).
2018-04-25 13:54:38,071 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-1266. Too few taxa (N < 3).
2018-04-25 13:54:38,071 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-1046. Too few taxa (N < 3).
2018-04-25 13:54:38,071 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-427. Too few taxa (N < 3).
2018-04-25 13:54:38,071 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-63. Too few taxa (N < 3).
2018-04-25 13:54:38,072 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-67. Too few taxa (N < 3).
2018-04-25 13:54:38,072 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-149. Too few taxa (N < 3).
2018-04-25 13:54:38,072 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-288. Too few taxa (N < 3).
2018-04-25 13:54:38,072 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-692. Too few taxa (N < 3).
2018-04-25 13:54:38,072 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-780. Too few taxa (N < 3).
2018-04-25 13:54:38,072 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-233. Too few taxa (N < 3).
2018-04-25 13:54:38,072 - phyluce_align_seqcap_align - WARNING - DROPPED locus uce-113. Too few taxa (N < 3).
2018-04-25 13:54:38,075 - phyluce_align_seqcap_align - INFO - Aligning with MAFFT
2018-04-25 13:54:38,075 - phyluce_align_seqcap_align - INFO - Alignment begins. 'X' indicates dropped alignments (these are reported after alignment)
.........................X....X.....X...................X...............X.......X................X......................................................X.........X..............X..........................................................................X.......................................X.........................................................................................................X...............................X...................................X...
2018-04-25 13:55:45,151 - phyluce_align_seqcap_align - INFO - Alignment ends
2018-04-25 13:55:45,151 - phyluce_align_seqcap_align - INFO - Writing output files
2018-04-25 13:55:45,193 - phyluce_align_seqcap_align - WARNING - DROPPED uce-852 from output
2018-04-25 13:55:45,194 - phyluce_align_seqcap_align - WARNING - DROPPED uce-1147 from output
2018-04-25 13:55:45,196 - phyluce_align_seqcap_align - WARNING - DROPPED uce-727 from output
2018-04-25 13:55:45,216 - phyluce_align_seqcap_align - WARNING - DROPPED uce-574 from output
2018-04-25 13:55:45,228 - phyluce_align_seqcap_align - WARNING - DROPPED uce-99 from output
2018-04-25 13:55:45,229 - phyluce_align_seqcap_align - WARNING - DROPPED uce-1188 from output
2018-04-25 13:55:45,352 - phyluce_align_seqcap_align - WARNING - DROPPED uce-752 from output
2018-04-25 13:55:45,476 - phyluce_align_seqcap_align - WARNING - DROPPED uce-47 from output
2018-04-25 13:55:45,590 - phyluce_align_seqcap_align - WARNING - DROPPED uce-668 from output
2018-04-25 13:55:45,591 - phyluce_align_seqcap_align - WARNING - DROPPED uce-1115 from output
2018-04-25 13:55:45,717 - phyluce_align_seqcap_align - WARNING - DROPPED uce-1129 from output
2018-04-25 13:55:45,841 - phyluce_align_seqcap_align - WARNING - DROPPED uce-904 from output
2018-04-25 13:55:45,881 - phyluce_align_seqcap_align - WARNING - DROPPED uce-1167 from output
2018-04-25 13:55:45,886 - phyluce_align_seqcap_align - WARNING - DROPPED uce-733 from output
2018-04-25 13:55:45,932 - phyluce_align_seqcap_align - WARNING - DROPPED uce-161 from output
2018-04-25 13:55:45,945 - phyluce_align_seqcap_align - INFO - ============== Completed phyluce_align_seqcap_align =============

```



Get summary stats
```{bash, evaluate=F}
phyluce_align_get_align_summary_data \
    --alignments ./embjac/alignments \
    --cores 4 \
    --log-path log
    
    
2018-04-25 13:57:09,292 - phyluce_align_get_align_summary_data - INFO - ========= Starting phyluce_align_get_align_summary_data =========
2018-04-25 13:57:09,299 - phyluce_align_get_align_summary_data - INFO - Version: 1.5.0
2018-04-25 13:57:09,300 - phyluce_align_get_align_summary_data - INFO - Argument --alignments: /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/embjac/alignments
2018-04-25 13:57:09,300 - phyluce_align_get_align_summary_data - INFO - Argument --cores: 4
2018-04-25 13:57:09,300 - phyluce_align_get_align_summary_data - INFO - Argument --input_format: nexus
2018-04-25 13:57:09,300 - phyluce_align_get_align_summary_data - INFO - Argument --log_path: /Users/cran5048/BaseSpace/UCE_Fish_2/phyluce/log
2018-04-25 13:57:09,300 - phyluce_align_get_align_summary_data - INFO - Argument --show_taxon_counts: False
2018-04-25 13:57:09,300 - phyluce_align_get_align_summary_data - INFO - Argument --verbosity: INFO
2018-04-25 13:57:09,300 - phyluce_align_get_align_summary_data - INFO - Getting alignment files
2018-04-25 13:57:09,303 - phyluce_align_get_align_summary_data - INFO - Computing summary statistics using 4 cores
2018-04-25 13:57:09,499 - phyluce_align_get_align_summary_data - INFO - ----------------------- Alignment summary -----------------------
2018-04-25 13:57:09,500 - phyluce_align_get_align_summary_data - INFO - [Alignments] loci:	455
2018-04-25 13:57:09,500 - phyluce_align_get_align_summary_data - INFO - [Alignments] length:	55,387
2018-04-25 13:57:09,500 - phyluce_align_get_align_summary_data - INFO - [Alignments] mean:	121.73
2018-04-25 13:57:09,500 - phyluce_align_get_align_summary_data - INFO - [Alignments] 95% CI:	5.16
2018-04-25 13:57:09,500 - phyluce_align_get_align_summary_data - INFO - [Alignments] min:	50
2018-04-25 13:57:09,500 - phyluce_align_get_align_summary_data - INFO - [Alignments] max:	471
2018-04-25 13:57:09,501 - phyluce_align_get_align_summary_data - INFO - ------------------------- Taxon summary -------------------------
2018-04-25 13:57:09,501 - phyluce_align_get_align_summary_data - INFO - [Taxa] mean:		5.90
2018-04-25 13:57:09,501 - phyluce_align_get_align_summary_data - INFO - [Taxa] 95% CI:	0.19
2018-04-25 13:57:09,501 - phyluce_align_get_align_summary_data - INFO - [Taxa] min:		3
2018-04-25 13:57:09,501 - phyluce_align_get_align_summary_data - INFO - [Taxa] max:		13
2018-04-25 13:57:09,502 - phyluce_align_get_align_summary_data - INFO - ----------------- Missing data from trim summary ----------------
2018-04-25 13:57:09,502 - phyluce_align_get_align_summary_data - INFO - [Missing] mean:	9.63
2018-04-25 13:57:09,502 - phyluce_align_get_align_summary_data - INFO - [Missing] 95% CI:	0.51
2018-04-25 13:57:09,502 - phyluce_align_get_align_summary_data - INFO - [Missing] min:	0.65
2018-04-25 13:57:09,502 - phyluce_align_get_align_summary_data - INFO - [Missing] max:	28.80
2018-04-25 13:57:09,507 - phyluce_align_get_align_summary_data - INFO - -------------------- Character count summary --------------------
2018-04-25 13:57:09,507 - phyluce_align_get_align_summary_data - INFO - [All characters]	308,871
2018-04-25 13:57:09,507 - phyluce_align_get_align_summary_data - INFO - [Nucleotides]		273,760
2018-04-25 13:57:09,507 - phyluce_align_get_align_summary_data - INFO - ---------------- Data matrix completeness summary ---------------
2018-04-25 13:57:09,508 - phyluce_align_get_align_summary_data - INFO - [Matrix 50%]		155 alignments
2018-04-25 13:57:09,508 - phyluce_align_get_align_summary_data - INFO - [Matrix 55%]		100 alignments
2018-04-25 13:57:09,508 - phyluce_align_get_align_summary_data - INFO - [Matrix 60%]		100 alignments
2018-04-25 13:57:09,508 - phyluce_align_get_align_summary_data - INFO - [Matrix 65%]		55 alignments
2018-04-25 13:57:09,508 - phyluce_align_get_align_summary_data - INFO - [Matrix 70%]		55 alignments
2018-04-25 13:57:09,508 - phyluce_align_get_align_summary_data - INFO - [Matrix 75%]		31 alignments
2018-04-25 13:57:09,508 - phyluce_align_get_align_summary_data - INFO - [Matrix 80%]		15 alignments
2018-04-25 13:57:09,508 - phyluce_align_get_align_summary_data - INFO - [Matrix 85%]		15 alignments
2018-04-25 13:57:09,508 - phyluce_align_get_align_summary_data - INFO - [Matrix 90%]		3 alignments
2018-04-25 13:57:09,508 - phyluce_align_get_align_summary_data - INFO - [Matrix 95%]		1 alignments
2018-04-25 13:57:09,509 - phyluce_align_get_align_summary_data - INFO - ------------------------ Character counts -----------------------
2018-04-25 13:57:09,509 - phyluce_align_get_align_summary_data - INFO - [Characters] '-' is present 3,859 times
2018-04-25 13:57:09,509 - phyluce_align_get_align_summary_data - INFO - [Characters] '?' is present 31,252 times
2018-04-25 13:57:09,509 - phyluce_align_get_align_summary_data - INFO - [Characters] 'A' is present 74,274 times
2018-04-25 13:57:09,509 - phyluce_align_get_align_summary_data - INFO - [Characters] 'C' is present 63,087 times
2018-04-25 13:57:09,509 - phyluce_align_get_align_summary_data - INFO - [Characters] 'G' is present 61,946 times
2018-04-25 13:57:09,509 - phyluce_align_get_align_summary_data - INFO - [Characters] 'T' is present 74,453 times
2018-04-25 13:57:09,509 - phyluce_align_get_align_summary_data - INFO - ========= Completed phyluce_align_get_align_summary_data ========

```

Create consensus sequences of all alignments in Geneious

Map reads to fasta
```{bash, evaluate=F}
phyluce_snp_bwa_align \
    --config ./embjac/embjac_phasing_conf.txt \
    --output ./embjac/mapping \
    --subfolder split-adapter-quality-trimmed \
    --log-path ./embjac/bwa_log \
    --cores 4
    
    
    
(phyluce) M053-303-26932:mapping2 cran5048$ mkdir bams
(phyluce) M053-303-26932:mapping2 cran5048$ find . -name "*.bam" -exec cp {} ./bams \;
(phyluce) M053-303-26932:mapping2 cran5048$ cd bams

for bam in *.bam
do
samtools index $bam
done

samtools faidx ../../contigs_consensus.fasta

(phyluce) M053-303-26932:bams cran5048$ samtools mpileup -uf ../../contigs_consensus.fasta *.bam | bcftools view -bvcgI  - > var.raw.bcf
[mpileup] 17 samples in 17 input files
<mpileup> Set max per-file depth to 470
[afs] 0:1401.204 1:36.449 2:24.606 3:16.974 4:12.449 5:9.728 6:7.988 7:6.800 8:5.942 9:5.289 10:4.767 11:4.334 12:3.971 13:3.672 14:3.424 15:3.209 16:3.007 17:2.797 18:2.575 19:2.346 20:2.124 21:1.923 22:1.751 23:1.609 24:1.497 25:1.412 26:1.354 27:1.326 28:1.333 29:1.385 30:1.501 31:1.721 32:2.134 33:2.923 34:4.475


 bcftools view var.raw.bcf | vcfutils.pl varFilter -d5 -Q30 -D100 > var.flt4.vcf
 
 11 SNPS
```

 
 
